obsreport/obsreporttest/deprecated.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package obsreporttest // import "go.opentelemetry.io/collector/obsreport/obsreporttest"
+
+import (
+	"go.opentelemetry.io/collector/component"
+	ortest "go.opentelemetry.io/collector/internal/obsreportconfig/obsreporttest"
+)
+
+// Deprecated: [0.85.0] Use go.opentelemetry.io/collector/internal/obsreportconfig/obsreporttest.TestTelemetry instead.
```

Comment:
`go.opentelemetry.io/collector/internal/obsreportconfig/obsreporttest ` cannot be used from the outside, it's internal

Reply:
do you have any suggestions where this should live? i dont really want to keep `obsreport` just for a testing subpackage

Reply:
`TestTelemetry` maybe can go to `componenttest` along with `SetupTelemetry`, but all its methods have to go to `[exporter|receiver|processor]test` modules as functions, similar to `CheckScraperMetrics`, which should go to `receivertest` or new `scrapertest` package I believe

Reply:
Btw seems like `CheckScraperMetrics` is not being used. It would be a nice to have an issue to whether adopt it or remove it.

Reply:
Hmmm... you're right it doesn't look like it's used outside of the package itself... i'll mark it as deprecated and we can remove it next release after that unless there's any pushback

Reply:
nm it's used in the scraperhelper. i'll move it to `componenttest` as well

Reply:
It cannot go to `componenttest` because is depends on `receiver`. Should go to `receivertest` or `scrapertest`